lib/identity_cache/configuration_dsl.rb

```diff
@@ -205,11 +205,20 @@ def build_recursive_association_cache(association, options) #:nodoc:
         options[:cached_accessor_name]   = "fetch_#{association}"
         options[:records_variable_name]  = "cached_#{association}"
 
-        self.class_eval(<<-CODE, __FILE__, __LINE__ + 1)
-          def #{options[:cached_accessor_name]}
-            fetch_recursively_cached_association('#{options[:records_variable_name]}', :#{association})
-          end
-        CODE
+        if method_defined?(options[:cached_accessor_name].to_sym)
+          self.class_eval(<<-CODE, __FILE__, __LINE__ + 1)
+            def #{options[:cached_accessor_name]}_with_idc
```

Comment:
this warning is `warning: method redefined; discarding old fetch_polymorphic_records`

Exists `#fetch_polymorphic_records` by AR.
`self.class_eval` is `#fetch_polymorphic_records` redefined.

This approach 

Define method(suffix `with_idc`), if exists method.
It using by alias_method_chain.